### PR TITLE
Make the server compile with the android gradle plugin

### PR DESCRIPTION
**Pull request by AndreyMitsyk**
**What is it?**
These changes allows us to compile calabash-android-server easily without taking care of multiple dependencies like specific version of android build tools, android platform, shared libs in repo etc.
Gradle file is simple right now.
Repo was cleaned up (many thanks to Simon for this 👍).
In addition we can use correct way to set target/min sdk in gradle file. This is important because we need an easy way to recompile app for specific configuration.

**Changes:**
- Moved [changes from Simon](https://github.com/calabash/calabash-android-server/commit/4638736dada0e4e80736a5b53e9ed74cf7a8004d) to common "server" folder.
- Fixed build steps after enabling android plugin.
- Implemented logic to set version during build steps without creation of temp folders like "staging".
- Changed build script to execute new gradle build and verify gradle file instead of second manifest.
- Changed server (example) manifest.
- Added removing of temp AndroidManifest and META-INF from result apk as a part of build script. This is needed to re-sign TestServer apk file.
- Implemented adding of sample manifest into apk assets.
- Changed logic of compiling calabash classes and tests.
- Updated Travis config.
- Updated gitignore files.

---

Tested updated server apk locally (integration tests with updated local gem + Xamarin.UITests).
Appcenter tests are passing for updated server.

### CODE REVIEW

File: bin/build.sh [hunk 1]
```diff
@@ -126,7 +97,10 @@ info "Built server ${CALABASH_ANDROID_SERVER_VERSION} with command:"
 
 shell "${CMD}"
 
-info "using ${ANDROID_TOOLS_DIR}"
-
 info "android:targetSdkVersion=${SERVER_API_LEVEL}"
 info "   android:minSdkVersion=${SERVER_MIN_VERSION}"
+
+zip -d "${APK_PATH}" META-INF/\*
+zip -d "${APK_PATH}" AndroidManifest.xml
```
Comment (kamstrup) [inline bin/build.sh]: Can you add a `-q` to zip to make it shut up :-) And then `echo "Stripping signature and manifest from server apk..."` instead.

File: bin/build.sh [hunk 2]
```diff
@@ -126,7 +97,10 @@ info "Built server ${CALABASH_ANDROID_SERVER_VERSION} with command:"
...
+
+mv "${APK_PATH}" "${SERVER_APK_PATH}"
```
Comment (kamstrup) [inline bin/build.sh]: Please add
```
echo "Build done."
echo "Output ready in: $SERVER_APK_PATH"
```

File: server/app/src/androidTest/java/sh/calaba/instrumentationbackend/actions/version/Version.java [hunk 3]
```diff
@@ -13,10 +13,10 @@
 
 	/******
 	 * Generate Version Number
-	 * DO NOT EDIT
+	 * DO NOT EDIT THIS FILE MANUALLY
 	 *
 	 */
-	public static final String VERSION="####VERSION####";
+	public static final String VERSION="0.9.2";
```
Comment (kamstrup) [inline server/app/src/androidTest/java/sh/calaba/instrumentationbackend/actions/version/Version.java]: ?! Do we intend to have a hardcoded version committed to git?

Comment (AndreyMitsyk) [reply server/app/src/androidTest/java/sh/calaba/instrumentationbackend/actions/version/Version.java]: This version will be updated during build

Comment (AndreyMitsyk) [reply server/app/src/androidTest/java/sh/calaba/instrumentationbackend/actions/version/Version.java]: task replaceVersion will update this. Anyway this version can't break anything and the fact, that we have version in source files allows us to get rid of "staging" folder

File: bin/build.sh (hunk 2 above)
Comment (AndreyMitsyk) [reply bin/build.sh]: done

File: bin/build.sh (hunk 1 above)
Comment (AndreyMitsyk) [reply bin/build.sh]: done

**Comment by kamstrup** [comment]
Good stuff Andrey. Ideally I would like to get rid of the bin/build.sh script entirely. The version juggling seems like it should all be easily doable from inside the gradle script... ? How far are we from being able to delete bin/build.sh?

**Comment by AndreyMitsyk** [comment]
@kamstrup I see that we can use gradle only. In this case we need to change CI logic a bit and move several steps to gradle. Right now build script is a wrapper which allow to verify some configuration before gradle build + it's a wrapper: we executes `bin/build.sh` instead of `server/gradlew clean assembleAndroidTest`. If it's ok, I can change logic and remove build.sh

**Comment by AndreyMitsyk** [comment]
@kamstrup Removed build script and changed CI configurations (Travis and Jenkins)

File: server/app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    }
+}
+
+task notifyBuildStarted {
+    doLast {
+        println("Build started...")
+    }
+}
+
+task replaceVersion {
```
Comment (john7doe) [inline server/app/build.gradle]: I'd love to get get rid of this replacing in src files step. So how about getting rid of the ../../version file and use $(versionName). Write the value of $(versionName) to a resource file and read rewrite the version action to return the value read from the file

Comment (AndreyMitsyk) [reply server/app/build.gradle]: I will keep original version file cause we are using this to create git tags as well. In addition, version in gradle is not aligned with common version.
Old version file will be copied into assets and we can read version in runtime.

**Pull request merged**